pkg/operator/ceph/object/user/controller.go

```diff
@@ -191,7 +191,22 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 
 	user, rgwerr, err := object.CreateUser(objContext, userConfig)
 	if err != nil {
-		return fmt.Errorf("failed to create user %s. RadosGW returned error %d: %+v", u.Name, rgwerr, err)
+		if rgwerr == object.RGWErrorBadData {
+			return fmt.Errorf("failed to create user %s. RadosGW returned error %d: %+v", u.Name, rgwerr, err)
```

Comment:
`return fmt.Errorf("failed to create rgw user %q. error code %d. %+v", u.Name, rgwerr, err)`